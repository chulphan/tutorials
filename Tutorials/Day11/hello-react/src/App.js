/*

  import를 하는 것은 webpack을 사용하기에 가능한 작업.
  나중에 프로젝트를 build하게 됐을때 webpack에서 파일의 확장자에 따라서 다른 작업을 하게 됨.
  CSS파일을 불러오면 프로젝트에서 사용한 프로젝트를 한 파일에 모두 결합해주는 작업을 진행.
  JS파일을 불러오면 모든 코드들이 제대로 로딩되게 순서를 설정하고 하나의 파일로 합쳐줌.
  사전에 따로 설정하지 않은 확장자(e.g, svg)의 경우 파일로서 불러온 다음에 나중에 특정경로에 사본을 만들어주고 해당 사본의 경로를 텍스트로 받아옴.

*/

// 파일에서 JSX를 사용하려면 React를 꼭 import 해주어야 함.
import React, { Component } from 'react';
import './App.css'

/*
  컴포넌트를 만드는 두가지 방법
  1. 클래스를 통해서(아래와 같이)
  2. 함수를 통해서.

  클래스 형태로 만들어진 컴포넌트는 반드시 render함수를 포함해야하고 render함수 내부에는 반드시 JSX를 return 해주어야 함.
  return 문 내에 있는 코드가 JSX형태임.
*/

/*
  div 두개를 감싸려고 div를 의미없이 쓰기 싫으면 Fragment라는 태그로 감싸주면 됨
  React.Fragment 로 사용.
*/

const state = function(value) {
  if(value === 1) return (<div>하나</div>);
  if(value === 2) return (<div>둘</div>);
  if(value === 3) return (<div>셋</div>);
}

class App extends Component {
  
  // 두번째 div와 같이 복잡한 조건을 작성해야 할 때는 JSX 밖에서 로직을 작성하는 것이 좋음.
  // 하지만 꼭 JSX 내부에서 작성해야 한다면 즉시호출함수를 사용해야함.

  // 리액트에서는 style을 지정할 때 객체형태로 작성해주어야 함.

  render() {
    const value = 3;
    const style = {
      backgroundColor: 'black',
      padding: '16px',
      color: 'white',
      fontSize: '12px'
    }
    return (
      <React.Fragment>
        <div>
          {
            1 + 1 === 2 ?
            (<div>맞아요!</div>) :
            (<div>틀료요!</div>)
          }
        </div>
        <div>
          {
            (function() {
              if(value === 1) return (<div>하나</div>);
              if(value === 2) return (<div>둘</div>);
              if(value === 3) return (<div>셋</div>);
            })()
          }
        </div>
        <div>
          {
            state(value)
          }
        </div>
        <div>
          {
            (() => {
              if(value === 1) return (<div>하나</div>);
              if(value === 2) return (<div>둘</div>);
              if(value === 3) return (<div>셋</div>);
            })()
          }
        </div>
        <div style={style}>
          STYLE!!
        </div>
        <div className="App">
        {/* 주석은 이렇게 태그 사이에... 그리고 리액트에서 클래스 이름은 className 으로 지정해준다. */}
          REACT!!!
        </div>
      </React.Fragment>
    );
  }
}

// 우리가 작성한 컴포넌트를 다른 곳에서 부를 수 있도록 내보내기를 해주는 것임.
export default App;
