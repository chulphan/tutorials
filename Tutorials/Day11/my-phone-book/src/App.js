import React, { Component } from 'react';
import PhoneForm from './components/PhoneForm';
import PhoneInfoList from './components/PhoneInfoList';

class App extends Component {

  id = 3

  state = {
    informations: [
      {
        id: 0,
        name: '철환',
        phone: '010-8579-1234'
      },
      {
        id: 1,
        name: '병철',
        phone: '010-5056-1234'
      },
      {
        id: 2,
        name: '빠돼',
        phone: '010-5453-1234'
      }
    ],
    keyword: ''
  }

  handleOnCreate = (data) => {
    const { informations } = this.state;
    this.setState ({
      informations: informations.concat({ id: this.id++, ...data })
    });
  }

  handleOnRemove = (key) => {
    const { informations } = this.state;
    this.setState({
      informations: informations.filter(({id}) => key !== id)
    })
  }

  handleOnUpdate = (id, data) => {
    const { informations } = this.state;
    this.setState({
      informations: informations.map((item) => item.id === id ? {...item, ...data} : item)
    })
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    })
  }

  render() {
    const { informations, keyword } = this.state;
    const filteredList = informations.filter((info) => info.name.indexOf(keyword) !== -1)
    return (
      <React.Fragment>
        <PhoneForm 
          onCreate={this.handleOnCreate}
        />
        <p>
          <input type="text"
                 value={this.state.keyword}
                 name="keyword"
                 onChange={this.handleChange}
          />
        </p>
        <PhoneInfoList 
          info={filteredList}
          onRemove={this.handleOnRemove}
          onUpdate={this.handleOnUpdate}
        />
      </React.Fragment>
    );
  }
}

export default App;
