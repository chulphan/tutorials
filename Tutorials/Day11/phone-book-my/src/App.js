import React, { Component } from 'react';
import PhoneForm from './components/PhoneForm';
import PhoneInfoList from './components/PhoneInfoList';

class App extends Component {

  id = 0
  state = {
    information: [],
    keyword: ''
  }

  onCreate = (data) => {
    const { information } = this.state;
    this.setState({
      information: information.concat({id: this.id++, ...data})
    })
  }

  onRemove = (id) => {
    const { information } = this.state;
    this.setState({
      information: information.filter((info) => info.id !== id)
    })
  }

  onUpdate = (id, data) => {
    const { information } = this.state;
    this.setState({
      information: information.map((info) => info.id === id ? {...info, ...data} : info)
    })
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    })
  }

  render() {
 
    const { information, keyword } = this.state;
    const filterList = information.filter((info) => info.name.indexOf(keyword) !== -1)
    return (
      <div>
        <PhoneForm 
          onCreate={ this.onCreate }
        />
        <p>
          <input type="text"
                 value={ keyword }
                 onChange={ this.handleChange }
                 placeholder="이름 검색"
                 name="keyword"
          />     
        </p>
        <PhoneInfoList 
          data={ filterList }
          onRemove={this.onRemove}
          onUpdate={this.onUpdate}
        />
      </div>
    );
  }
}

export default App;
