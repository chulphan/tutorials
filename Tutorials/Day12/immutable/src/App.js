import React, { Component } from 'react';
import UserList from './UserList';
import { Map, List, Record } from 'immutable';

const User = Record({
  id: null,
  username: null
});

const Data = Record({
  input: '',
  users: List()
});

class App extends Component {
  
  
  id = 3;

  // state = {
  //   input: '',
  //   users: [
  //     {
  //       id: 1,
  //       username: 'velopert'
  //     },
  //     {
  //       id: 2,
  //       username: 'mjkim'
  //     }
  //   ]
  // }

  // state = {
  //   data: Map({
  //     input: '',
  //     users: List([
  //       Map({
  //         id: 1,
  //         username: 'velopert'
  //       }),
  //       Map({
  //         id: 2,
  //         username: 'chulhwan'
  //       })
  //     ])
  //   })
  // }

  state = {
    data: Data({
      users: List([
        User({
          id: 1,
          username: 'antil'
        }),
        User({
          id: 2,
          username: 'chkim'
        })
      ])
    })
  }

  onChange = (e) => {
    const { value } = e.target;
    const { data } = this.state;
    // this.setState({
    //   input: value
    // });
    this.setState({
      data: data.set('input', value)
    });
  }

  onButtonClick = (e) => {
    const { data } = this.state;
    // this.setState(({ users, input }) => ({
    //   data: data.set('input', '')
    //             .update('users', users => users.push(Map({
    //               id: this.id++,
    //               username: data.get('input')
    //             })))
    // }))

    this.setState({
      data: data.set('input', '')
                .update('users', users => users.push(new User({
                  id: this.id++,
                  username: data.get('input')
                })))
    })
  }

  render() {
    const { onChange, onButtonClick } = this;
    const { data: { input, users } } = this.state;

    return (
      <div>
        <div>
          <input onChange={onChange} value={input} />
          <button onClick={onButtonClick}>추가</button>
        </div>
        <h1>사용자 목록</h1>
        <div>
          <UserList users={users} />
        </div>
      </div>
    );
  }
}

export default App;