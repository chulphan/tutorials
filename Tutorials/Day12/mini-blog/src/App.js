import React, { Component } from 'react';
import BlogList from './components/blog/BlogList';
import LikeList from './components/like/LikeList';
import data from './data/data.json';

class App extends Component {

  likeId = 0

  state = {
    bloglist: [],
    likelist: [],
    displayed: false
  }

  componentDidMount() {

  }

  handleDisplayBlog = () => {
    const { displayed } = this.state;


    if (displayed) {
      this.setState({
        bloglist: [],
        displayed: !displayed
      });
    } else {
      this.setState({
        bloglist: [...data.body],
        displayed: !displayed
      });
    }
  }

  addLikeList = (data) => {
    const { likelist } = this.state;
  
    this.setState({
      likelist: likelist.concat({ id: this.likeId++, title: data })
    });
  }

  removeLikeList = (title) => {
    const { likelist } = this.state;
    this.setState({
      likelist: likelist.filter((like) => like.title !== title)
    });
  }

  render() {
    const { bloglist, likelist } = this.state;
    console.log('blog list', bloglist);
    return (
      <div>
        <h1>MINI BLOG!!!</h1>
        <button onClick={this.handleDisplayBlog}>글 불러오기</button>
        <BlogList list={bloglist}
          addLikeList={this.addLikeList}
        />
        <LikeList
          likes={likelist}
          removeLikeList={this.removeLikeList}
        />
      </div>
    );
  }
}

export default App;
