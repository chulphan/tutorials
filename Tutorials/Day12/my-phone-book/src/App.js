import React, { Component } from 'react';
import PhoneForm from './components/PhoneForm'
import PhoneInfoList from './components/PhoneInfoList';

class App extends Component {
  
  id = 0
  state = {
    phonelist: [],
    keyword: ''
  }

  onCreateBook = (data) => {
    const { phonelist } = this.state;
    this.setState({
      phonelist: phonelist.concat({id: this.id++, ...data})
    });
  }

  onRemoveBook = (id) => {
    const { phonelist } = this.state;
    this.setState({
      phonelist: phonelist.filter((list) => list.id !== id)
    });
  }

  onUpdateBook = (id, data) => {
    const { phonelist } = this.state;
    this.setState({
      phonelist: phonelist.map((list) => {
        return list.id === id ? {...list, ...data} : list
      })
    });
  }

  handleChange = (e) => {
    const { name, value } = e.target;

    this.setState({
      [name]: value
    })
  }

  render() {
    const { phonelist, keyword } = this.state;
    const filterPhoneList = phonelist.filter((list) => list.name.indexOf(keyword) !== -1)
    return (
      <div>
        <PhoneForm 
          onCreateBook={ this.onCreateBook }
        /> 
        <p>
          <input type="text"
                 value={ keyword }
                 name="keyword"
                 onChange={ this.handleChange }
                 placeholder="이름 검색"
          />
        </p>
        <PhoneInfoList
          list={ filterPhoneList }
          onRemoveBook={ this.onRemoveBook }
          onUpdateBook={ this.onUpdateBook }
        />
      </div>
    );
  }
}

export default App;
