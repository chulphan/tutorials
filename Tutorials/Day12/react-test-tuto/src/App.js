import React, { Component } from 'react';
import Counter from './counter/Counter';
import NameForm from './name/NameForm';
import NameList from './name/NameList';


class App extends Component {
  id = 0
  state = {
    namelist: []
  }

  onCreate = (data) => {
    const { namelist } = this.state;

    this.setState({
      namelist: namelist.concat({id: this.id++, ...data})
    });
  }

  render() {
    return (
      <div>
        <Counter />
        <NameForm
          onCreate={this.onCreate}
        />
        <NameList
          namelist={this.state.namelist}
        />
      </div>
    );
  }
}

export default App;
