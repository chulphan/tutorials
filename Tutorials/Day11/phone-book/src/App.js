import React, { Component } from 'react';
import './App.css';
import PhoneForm from './components/PhoneForm';
import PhoneInfoList from './components/PhoneInfoList';

class App extends Component {
  id = 2
  state = {
    information: [
      {
        id: 0,
        name: '김철환',
        phone: '010-8579-5014'
      },
      {
        id: 1,
        name: '문수',
        phone: '010-1234-1234'
      }
    ],
    keyword: ''
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    console.log(e);
    this.setState({
      [name]: value
    })
  }

  handleCreate = (data) => {
    const { information } = this.state;
    this.setState({
      information: information.concat({ id: this.id++, ...data})
    }) 
  }

  handleRemove = (id) => {
    const { information } = this.state;
    this.setState({
      information: information.filter(info => info.id !== id) 
    })
  }

  handleUpdate = (id, data) => {
    const { information } = this.state;
    this.setState({
      information: information.map((info) => info.id === id ? ({...info, ...data}) : info)
    })
  }

  render() {
    const { information, keyword } = this.state;
    const filteredList = information.filter(
      info => info.name.indexOf(keyword) !== -1
    )
    return (
      <div>
        <PhoneForm
          onCreate={this.handleCreate}
        />
        <PhoneInfoList 
          data={filteredList} 
          onRemove={this.handleRemove}
          onUpdate={this.handleUpdate}
        />
        <input type="text"
               name="keyword"
               placeholder="이름을 입력해주세요"
               value={this.state.keyword}
               onChange={this.handleChange}
        />
      </div>
    );
  }
}

export default App;
