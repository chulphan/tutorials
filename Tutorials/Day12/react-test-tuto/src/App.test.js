import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from './App';



// 해당 컴포넌트가 크래쉬 없이 제대로 렌더링 되었는지 확인해줌.
describe('App', () => {
  let component = null;

  it('renders correctly', () => {
    component = renderer.create(<App/>);
  });

  it('matches snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
