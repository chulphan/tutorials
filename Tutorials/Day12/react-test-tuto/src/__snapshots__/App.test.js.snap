// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div>
  <div>
    <h1>
      카운터
    </h1>
    0
    <br />
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        name="name"
        onChange={[Function]}
        placeholder="이름을 입력해주세요"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        등록
      </button>
    </div>
  </form>
  <div />
</div>
`;
